### PR TITLE
Use atomic file move operations.

### PR DESCRIPTION
**Pull request by olafurpg**
Following up on https://github.com/scalameta/metals/pull/935#discussion_r354406977

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala [hunk 1]
```diff
@@ -344,7 +345,27 @@ object MetalsEnrichments
...
+      try {
+        Files.move(
+          tmp,
+          path.toNIO,
+          StandardCopyOption.REPLACE_EXISTING,
+          StandardCopyOption.ATOMIC_MOVE
+        )
+      } catch {
+        case NonFatal(_) =>
+          Files.move(tmp, path.toNIO, StandardCopyOption.REPLACE_EXISTING)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala]: So there is still a risk we will have a non-atomic operation? Shouldn't we retry the atomic move again or resign altogether.

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala]: Some file systems don’t support atomic moves so the best we can do is fall back to a normal move.

Comment (gabro) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala]: Could there be a more specific exception to catch here?

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala]: The docs don't specify what exception will be thrown so I didn't want to be too specific.

**Comment by olafurpg** [comment]
Ignoring unrelated CI failures.

```
 X tests.TreeViewLspSuite.no-op 3539ms 
  tests.TestFailedException: failed assertion at /Users/runner/runners/2.162.0/work/metals/metals/tests/unit/src/test/scala/tests/TreeViewLspSuite.scala:499
```

**Pull request merged**